.script/SolutionValidations/validDomainsVerticals.ts

```diff
@@ -0,0 +1,102 @@
+import { MainTemplateDomainVerticalValidationError } from "./../utils/validationError";
+import * as logger from "./../utils/logger";
+import { ExitCode } from "../utils/exitCode";
+import fs from "fs";
+
+// initialize arrays to store valid domains and verticals
+let validDomains: string[] = [];
+let validVerticals: string[] = [];
+
+// read the valid domains and verticals from the JSON file
+try {
+    const validDomainsVerticals = JSON.parse(fs.readFileSync('./.script/SolutionValidations/ValidDomainsVerticals.json', "utf8"));
+    validDomains = validDomainsVerticals.validDomains;
+    validVerticals = validDomainsVerticals.validVerticals;
+} catch (error) {
+    logger.logError(`Error reading ValidDomainsVerticals.json file: ${error}`);
+}
+
+// function to check if the solution is valid
+export function IsValidSolutionDomainsVerticals(filePath: string): ExitCode {
+
+    // check if the file is a mainTemplate.json file
+    if (filePath.endsWith("mainTemplate.json")) {
+        // read the content of the file
+        let jsonFile = JSON.parse(fs.readFileSync(filePath, "utf8"));
+
+        // check if the file has a "resources" field
+        if (!jsonFile.hasOwnProperty("resources")) {
+            console.warn(`No "resources" field found in the file. Skipping file path: ${filePath}`);
```

Comment:
Shouldn't be this error? Is there any scenario where you don't have resources in main template?

Reply:
Added the exception when there is no resource exists.